preload/vperm_filestatgates.c

```diff
@@ -899,7 +924,7 @@ int __xmknodat_gate(int *result_errno_ptr,
 
 		SB_LOG(SB_LOGLEVEL_DEBUG, "%s: real fn: %d, errno=%d",
 			__func__, res, e);
-		if (e == EPERM) {
+		if (errno == EPERM) {
```

Comment:
I could guess this change is not needed as e has the errno and logging should not change it. Also that is printed by the logger, or does the logger do something funny?

Reply:
The logger only logs the message to the logging output, i.e. the logging file. 
In case that would fail errno could change the relevant errno is the one stored in e not the one after the call to the logger.

The change would mean that if the errno changed after the call to the logger that the if check above would be wrong.

Reply:
Still, I do not see much sense in this change. I think the saved value of `errno` in `e` should be compared here?

Reply:
This was changed to use e instead of errno.